Download.Mvc/Client/components/FolderEntry.tsx

```diff
@@ -0,0 +1,58 @@
+﻿import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { iconStyle } from './FolderView';
+import { FileView } from './FileView';
+
+export class FolderEntry extends React.Component<any, any>{
+    constructor(props) {
+        super(props);
+        this.updateReadmeID = this.updateReadmeID.bind(this);
+        this.updateLicenseID = this.updateLicenseID.bind(this);
```

Comment:
@laholstege if you just use the props directly, you can get rid of the local functions entirely.  If you want to keep them for some reason, if you change the signature to `updateReadmeID = (newID) => { ... }` then you don't need this bind in the constructor at all